pkg/cluster/nodes/create.go

```diff
@@ -48,18 +48,23 @@ func getPort() (int, error) {
 
 // CreateControlPlaneNode creates a contol-plane node
 // and gets ready for exposing the the API server
-func CreateControlPlaneNode(name, image, clusterLabel string) (node *Node, err error) {
+func CreateControlPlaneNode(name, image, network, clusterLabel string) (node *Node, err error) {
 	// gets a random host port for the API server
 	port, err := getPort()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get port for API server")
 	}
 
-	node, err = createNode(name, image, clusterLabel, config.ControlPlaneRole,
+	extraArgs := []string{
 		// publish selected port for the API server
 		"--expose", fmt.Sprintf("%d", port),
 		"-p", fmt.Sprintf("%d:%d", port, kubeadm.APIServerPort),
-	)
+	}
+	if network != "" {
+		extraArgs = append(extraArgs, "--network", network)
```

Comment:
this binds a kind CLI flag to the underlying CR (docker).
i guess we need to evaluate if we want this coupling for the 1.0 release.

Reply:
yeah, i'm iffy on this. 🤔 
I this probably does make sense as a flag, I guess, but as a rule of thumb we should prefer config... 🤔

Reply:
If it was in the config would it be at the top level?
Is there any setup where your nodes would be in different networks?

Reply:
sorry, time flew by so fast :(

i'm not sure.. I could see this maybe, we could start top level and do an override.

Either way we're mostly decoupled from docker at the user interface right now, and I do intend to support podman etc., I meant to investigate further if there was a better way to specify this or any other work arounds.

I do see how this is a feature we need to enable.